lightning/src/chain/mod.rs

```diff
@@ -315,11 +315,12 @@ pub trait Watch<ChannelSigner: Sign> {
 /// [BIP 157]: https://github.com/bitcoin/bips/blob/master/bip-0157.mediawiki
 /// [BIP 158]: https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki
 pub trait Filter {
-	/// Registers interest in a transaction with `txid` and having an output with `script_pubkey` as
+	/// Registers interest in funding transactions to inform LDK that a channel
+	/// Funding transaction is transaction with `txid` and having an output with `script_pubkey` as
```

Comment:
I don't think we want to narrow the definition of this function to only funding transaction(s). Yes, it's true today, but specifying it in the docs and committing to it I don't think we want to do.

Reply:
What would be a better description that clarifies and differentiates what it does with register_output

Reply:
The key difference here is one looks at a given transaction being broadcasted, one looks for *spends* of a given output being broadcasted.

Reply:
@TheBlueMatt, should I go with this; 

/// Registers interest in transactions to inform LDK that a channel.
/// Transactions with `txid` and having an output with `script_pubkey` as
/// a spending condition.

Reply:
I dont think these functions should have any reference to what kind of transaction it is (channel/close/open/etc).

Reply:
So I should leave them at what they were?

Reply:
I mean I do think they need to be clarified, because we've seen a good chunk of confusion on them, but it remains entirely unclear to me exactly *how* they should be clarified :/